tensorflow/python/keras/layers/normalization.py

```diff
@@ -527,15 +528,18 @@ def _fused_batch_norm(self, inputs, training):
 
     original_shape = None
     fused_axis = self.axis[0]
-    input_shape = inputs.shape.as_list()
-    ndims = len(input_shape)
+    input_shape = array_ops.shape(inputs)
+    ndims = len(inputs.shape)
     if self.axis[0] not in (1, ndims - 1) or ndims != 4:
       # The fused implementation only supports NCHW or NHWC, so we reshape the
       # input/output tensor to/from an equivalent 4D shape.
       fused_shape, fused_axis = self._get_shape_and_axis_for_fused(input_shape,
                                                                    self.axis[0])
-      original_shape = [-1] + input_shape[1:]
-      inputs = array_ops.reshape(inputs, [-1] + fused_shape[1:])
+      original_shape = array_ops.concat(
+          [constant_op.constant([-1]), input_shape[1:]], axis=0)
+      fused_shape = array_ops.concat(
+          [constant_op.constant([-1]), fused_shape[1:]], axis=0)
```

Comment:
Why do you need to replace the first element of `fused_shape` with -1?

Reply:
Good point, I've removed these operations.